charts/tempo-distributed/templates/_helpers.tpl

```diff
@@ -34,6 +34,19 @@ Docker image selector for Tempo. Hierachy based on global, component, and tempo
 {{- printf "%s/%s:%s" $registry $repository $tag -}}
 {{- end -}}
 
+{{/*
+Optional list of imagePullSecrets for Tempo docker images
+*/}}
+{{- define "tempo.imagePullSecrets" -}}
+{{- $imagePullSecrets := coalesce .global.pullSecrets .component.pullSecrets .tempo.pullSecrets -}}
```

Comment:
looking for clarification on "global" vs "tempo" pullsecrets. is there a reason to have both?

Reply:
I actually started writing it for global/component only, but then I noticed you have Tempo images grouped in `tempo` key, e.g. here https://github.com/grafana/helm-charts/blob/c4fdafbeae595f51ff9f33dc3ebb8a34ec6deb39/charts/tempo-distributed/templates/_helpers.tpl#L31

so I wanted to preserve the same logic here, because I thought variable precedence is `component > tempo > global` but now I see it's actually `global > component > tempo` :thinking: and I don't really understand why.

I guess you do, so should I use this instead?
```suggestion
{{- $imagePullSecrets := coalesce .global.pullSecrets .component.pullSecrets -}}
```
and edit it in relevant places?